src/Content/Widget/CalendarExport.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * @file src/Content/Widget/CalendarExport.php
+ */
+
+namespace Friendica\Content\Widget;
+
+use Friendica\Content\Feature;
+use Friendica\Core\L10n;
+
+require_once 'boot.php';
+require_once 'include/text.php';
+
+/**
+ * TagCloud widget
+ *
+ * @author Rabuzarus
+ */
+class CalendarExport
+{
+	/**
+	 * @brief Get the events widget.
+	 *
+	 * @return string Formated HTML of the calendar widget.
+	 */
+	public static function getHTML() {
+		$a = get_app();
+
+		$owner_uid = $a->data['user']['uid'];
+
+		// The permission testing is a little bit tricky because we have to respect many cases.
+
+		// It's not the private events page (we don't get the $owner_uid for /events).
+		if (!local_user() && !$owner_uid) {
+			return;
+		}
+
+		/*
+		 * If it's a kind of profile page (intval($owner_uid)) return if the user not logged in and
+		 * export feature isn't enabled.
+		 */
+		if (!local_user() && $owner_uid && !Feature::isEnabled($owner_uid, 'export_calendar')) {
```

Comment:
Why did you remove the second check? Think of a `local_user()` who visit a foreign profile on the same server. Without the second check the `local_user()` could export the events of a foreign profile although the profile owner has disabled the calendar export

Reply:
According to the chart, the second check doesn't matter, it passes the same way as the first one.

Reply:
Please be clearer. I don't get it.

The first check ensures that the widget is only shown if it is a local system page (like `events`) or a profile page (like `cal` -- note: `$owner_uid` is only available on profile pages)

The second check ensures that the `local_user()` does access to the widget on foreign profile pages on the same server if the profile owner has disabled the widget. At the same time it ensures that the `local_user()` does have access to the export widget if it is his/her own profile page (`/cal` module).

The third check is the regular check which checks if the feature is enabled for visitors.

What do you mean with:
> For example I'm surprised we don't test the feature on its own, which would actually prevent the widget to show if the user disabled the feature.

The widget should appear for the local users `/events` and his/her own `/cal` page, regardless of whether it is enabled or not

Reply:
What I mean is that the original set of permissions checks probably weren't exactly doing what was expected from them, unless I messed up again in my spreadsheet, of course.

I believe my current version is logically equivalent to the original thanks to your help, but it might not be "right" as "not doing what was expected" but it would have been true of the original version as well.

I don't have a problem fixing whatever needs fixing if we can come up with a complete and intelligible set of permissions checks.

Reply:
It turns out I was wrong again, I decided to trust the machine rather than myself, and wrote formulas in my spreadsheet. It turns out I had a couple of mistakes in the 2nd check, which makes it actually cover the single case the 3rd check did.

I ended up keeping only two checks, and I verified it keeps the same truth table as before in the spreadsheet.